pallets/confidential-asset/src/lib.rs

```diff
@@ -90,19 +110,47 @@ impl EncryptedBalanceWrapper {
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, Default)]
 pub struct MercatAccount {
     pub encrypted_asset_id: EncryptedAssetIdWrapper,
-    pub encryption_pub_key: EncryptionPubKey,
+    pub encryption_pub_key: EncryptionPubKeyWrapper,
 }
 
 impl MercatAccount {
-    /// Unwraps the value so that it can be passed to meract library.
-    fn to_mercat<T: Trait>(&self) -> Result<PubAccount, Error<T>> {
+    /// Unwraps the value so that it can be passed to mercat library.
+    pub fn to_mercat<T: Trait>(&self) -> Result<PubAccount, Error<T>> {
         Ok(PubAccount {
             enc_asset_id: self.encrypted_asset_id.to_mercat()?,
-            owner_enc_pub_key: self.encryption_pub_key,
+            owner_enc_pub_key: self.encryption_pub_key.to_mercat()?,
         })
     }
 }
 
+/// Wrapper for the mercat account proof that correspond to base64 encoding of `PubAccountTx`.
+/// Since this is received as input from user and is a binary data, the `Vec<u8>` will be a base64 encoded.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default)]
+pub struct PubAccountTxWrapper(pub Vec<u8>);
+
+impl PubAccountTxWrapper {
+    /// Unwraps the value so that it can be passed to mercat library.
+    pub fn to_mercat<T: Trait>(&self) -> Result<PubAccountTx, Error<T>> {
+        let mut data: &[u8] =
+            &base64::decode(self.0.clone()).map_err(|_| Error::<T>::UnwrapMercatDataError)?;
+        PubAccountTx::decode(&mut data).map_err(|_| Error::<T>::UnwrapMercatDataError)
+    }
+}
+
+/// Wrapper for the asset issuance proof that correspond to `InitializedAssetTx`.
+/// Since this is received as input from user and is a binary data, the `Vec<u8>` will be a base64 encoded.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default)]
+pub struct InitializedAssetTxWrapper(pub Vec<u8>);
+
+impl InitializedAssetTxWrapper {
+    /// Unwraps the value so that it can be passed to mercat library.
+    pub fn to_mercat<T: Trait>(&self) -> Result<InitializedAssetTx, Error<T>> {
+        let mut data: &[u8] =
+            &base64::decode(self.0.clone()).map_err(|_| Error::<T>::UnwrapMercatDataError)?;
```

Comment:
As we discussed, you could have strongly typed base64-encoded `Vec<u8>`, let's say `struct VecU8Base64(pub Vec<u8>)`. For that type you could define a `From` instance that would base64-encode an input `Vec<u8>`.

This allows defining a `From` for `EncryptedBalanceWrapper` that takes a `VecU8Base64` as input and outputs `EncryptedBalanceWrapper`. This step always succeeds if I'm not mistaken, so `From` is sufficient (otherwise I suggest `TryFrom`).